modules/sr/robot/robot.py

```diff
@@ -151,7 +156,7 @@ def sleep(self, secs: float) -> None:
             raise ValueError('sleep length must be non-negative')
 
         # Ensure the time delay is a valid step increment
-        n_steps = int((secs * 1000) // self._timestep)
+        n_steps = int((secs * 1000) // self._timestep) or 1
```

Comment:
The alternative here would be to not do this and have a nicer error at this layer. I've not put a lot of thought into whether that's better here. (Certainly it has the disadvantage that teams may find that code which currently works stops working)

Reply:
Should we just error here if you can’t delay for this amount of time, rather than truncating the wait time down to the timestep?

Reply:
Alternatively: use an accumulator to minimise the error over time.

Reply:
00eb22fe6d4bf5838e5680ecf82650a4d31d9b66 resolves this by rounding up rather than down (which is still within the API contract of `time.sleep`).